client/lib/interval/runner.js

```diff
@@ -0,0 +1,97 @@
+import { fromJS } from 'immutable';
+
+export const EVERY_SECOND = 1000;
+export const EVERY_FIVE_SECONDS = 5 * 1000;
+export const EVERY_TEN_SECONDS = 10 * 1000;
+export const EVERY_THIRTY_SECONDS = 30 * 1000;
+export const EVERY_MINUTE = 60 * 1000;
+
+const initialState = fromJS( {
+	nextId: 1,
+	periodTimers: {
+		EVERY_SECOND: null,
+		EVERY_FIVE_SECONDS: null,
+		EVERY_TEN_SECONDS: null,
+		EVERY_THIRTY_SECONDS: null,
+		EVERY_MINUTE: null
+	},
+	actions: []
+} );
+let state = initialState;
+
+const increment = a => a + 1;
+const push = item => list => list.push( item );
+const pull = id => list => list.filterNot( o => o.get( 'id' ) === id );
+
+export const resetForTesting = () => {
+	state
+		.get( 'periodTimers' )
+		.forEach( clearTimeout );
+
+	state = initialState;
+};
+
+export function add( period, onTick ) {
+	const id = state.get( 'nextId' );
+
+	storeNewAction( { id, period, onTick } );
+	scheduleNextRun();
+
+	return id;
+}
+
+function storeNewAction( { id, period, onTick } ) {
+	state = state
+		.update( 'actions', push( fromJS( { id, period, onTick } ) ) )
+		.update( 'nextId', increment );
+
+	return state.get( 'nextId' );
```

Comment:
That was left over from earlier code when I used the return value, but split the functionality into two places to separate the act of storing an action from the act of scheduling an action. Removed in 5084f14